hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SingleTransactionRecordBuilderImpl.java

```diff
@@ -344,6 +362,35 @@ public SingleTransactionRecordBuilderImpl transactionID(@NonNull final Transacti
         return this;
     }
 
+    /**
+     * When we update nonce on the record, we need to update the body as well with the same transactionID.
+     * @return the builder
+     */
+    @NonNull
+    public SingleTransactionRecordBuilderImpl syncBodyIdFromRecordId() {
+        final var newTransactionID = transactionID;
+        try {
+            final var signedTransaction = SignedTransaction.PROTOBUF.parseStrict(
+                    transaction.signedTransactionBytes().toReadableSequentialData());
+            final var existingTransactionBody =
+                    TransactionBody.PROTOBUF.parse(signedTransaction.bodyBytes().toReadableSequentialData());
+            final var body = existingTransactionBody
+                    .copyBuilder()
+                    .transactionID(newTransactionID)
+                    .build();
+            final var newBodyBytes = TransactionBody.PROTOBUF.toBytes(body);
+            final var newSignedTransaction =
+                    SignedTransaction.newBuilder().bodyBytes(newBodyBytes).build();
+            final var signedTransactionBytes = SignedTransaction.PROTOBUF.toBytes(newSignedTransaction);
+            this.transaction = Transaction.newBuilder()
+                    .signedTransactionBytes(signedTransactionBytes)
+                    .build();
+            return this;
+        } catch (Exception e) {
```

Comment:
A few updates here for the transaction. Maybe we want to catch the specific exception and do something, log. If just throwing a RuntimeException, what's going to happen to the calling code if failed?

Reply:
I see we are throwing RuntimeException in one other place in this class, so chose to do that 🤔